File: leetcode/945/2.go
// 哈希表解决冲突 开发地址法
func minIncrementForUnique(A []int) int {
	var arr [8000]int
	res := 0
	for _,m := range A {
		for {
			if arr[m] == 0 {
				arr[m]++
				break
			} else {
				m++
				res++
			}
		}
	}
	return res
}

File: leetcode/945/2.js
var minIncrementForUnique = function(A) {
  A = A.sort((a,b) => a - b)
  let res = []
  let move = 0
  res.push(A[0])
  for (let i = 1; i < A.length; i++) {
    if (A[i] > res[i-1]) {
      res.push(A[i])
    } else {
      while (A[i] <= res[i - 1]) {
        A[i]++
        move++
      }
      res.push(A[i])
    }
  }
  return move
}

var A = [1,2,2]
// var A = [3,2,1,2,1,7]
console.log(minIncrementForUnique(A))